base/server/python/pki/server/cli/migrate.py

```diff
@@ -601,3 +602,45 @@ def create_link(self, instance, source, dest):
 
         os.symlink(source, dest)
         os.lchown(dest, instance.uid, instance.gid)
+
+    def migrate_service(self, instance):
+        self.migrate_service_java_home(instance)
+
+    def migrate_service_java_home(self, instance):
+        # When JAVA_HOME in the Tomcat service config differs from the
+        # value in /usr/share/pki/etc/pki.conf, update the value in
+        # the service config.
+
+        if "JAVA_HOME" not in os.environ or not os.environ["JAVA_HOME"]:
+            logger.debug("Refusing to migrate JAVA_HOME with missing environment variable")
+
+        java_home = os.environ["JAVA_HOME"]
```

Comment:
If we're removing JAVA_HOME, I think this shouldn't be needed anymore.

Reply:
We'll still want to keep lines 614 and 615 though, right? Otherwise, the service will use an auto-detected default when `JAVA_HOME` is removed.

Reply:
It probably also needs a return statement. 

Also, it'd be a faulty configuration if `pki-server migrate` gets run without `JAVA_HOME` being set, IMO.